openapi_python_client/openapi_parser/openapi.py

```diff
@@ -154,24 +154,32 @@ class Schema:
     relative_imports: Set[str]
 
     @staticmethod
-    def from_dict(d: Dict[str, Any], /) -> Schema:
-        """ A single Schema from its dict representation """
+    def from_dict(d: Dict[str, Any], /, name: str = None) -> Schema:
+        """ A single Schema from its dict representation
+        :param d:    Dict representation of the schema
+        :param name: Name by which the schema is referenced, such as a model name.  Used to infer the type name if a `title` property is not available.
+        """
         required_set = set(d.get("required", []))
         required_properties: List[Property] = []
         optional_properties: List[Property] = []
         relative_imports: Set[str] = set()
 
-        for key, value in d["properties"].items():
-            required = key in required_set
-            p = property_from_dict(name=key, required=required, data=value)
-            if required:
-                required_properties.append(p)
-            else:
-                optional_properties.append(p)
-            if isinstance(p, (ReferenceListProperty, EnumListProperty, RefProperty, EnumProperty)) and p.reference:
-                relative_imports.add(import_string_from_reference(p.reference))
+        ref = Reference.from_ref(d.get("title", name))
+
+        if "properties" in d:
+            for key, value in d["properties"].items():
+                required = key in required_set
+                p = property_from_dict(name=key, required=required, data=value)
+                if required:
+                    required_properties.append(p)
+                else:
+                    optional_properties.append(p)
+                if isinstance(p, (ReferenceListProperty, EnumListProperty, RefProperty, EnumProperty)) and p.reference:
+                    # don't add an import for self-referencing schemas
+                    if p.reference.class_name != ref.class_name:
+                        relative_imports.add(import_string_from_reference(p.reference))
```

Comment:
Just a personal preference, I try to avoid multiple layers as much as possible.  Something about "cognitive complexity" I read in a book 🧠🤓. Feel free to disagree and leave as is.

```suggestion
        for key, value in d.get("properties", {}).items():
            required = key in required_set
            p = property_from_dict(name=key, required=required, data=value)
            if required:
                required_properties.append(p)
            else:
                optional_properties.append(p)
            if (
                isinstance(p, (ReferenceListProperty, EnumListProperty, RefProperty, EnumProperty))
                and p.reference
                and p.reference.class_name != ref.class_name  # don't add an import for self-referencing schemas
            ):
                relative_imports.add(import_string_from_reference(p.reference))
```